File: src/0226-翻转二叉树/0226.py
class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if not root:
            return root
        
        q_ = [root]
        while q_:
            node = q_.pop()
            left = node.left
            right = node.right
            # 左右两个子节点交换
            node.left = right
            node.right = left
            if node.left:
                q_.append(node.left)
            if node.right:
                q_.append(node.right)
            
        return root

File: src/0226-翻转二叉树/0226.cpp
 /**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        invert(root);
        return root;
    }
private:
    void invert(TreeNode*& tree)
    {
        if(tree == nullptr)
            return;
        TreeNode* leftTree = tree->left;
        tree->left = tree->right;
        tree->right = leftTree;

        invert(tree->left);
        invert(tree->right);
    }
};